util/logutil/hex.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by aprettyPrintlicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logutil
+
+import (
+	"bytes"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"reflect"
+	"strings"
+
+	"github.com/golang/protobuf/proto"
+)
+
+// ToHex formats a proto.Message into a string with the []byte fields in hex encoding.
+func ToHex(msg proto.Message) string {
+	val := reflect.ValueOf(msg)
+	var w bytes.Buffer
+	prettyPrint(&w, val)
+	return w.String()
```

Comment:
how about return w, then we could still use `zap.Stringer` in somewhere.

Reply:
`zap.Stringer` can be used on something that implements the [`fmt.Stringer`](https://godoc.org/fmt#Stringer) interface.

We can't implement `String() string` for `metapb.Region` because it's defined in the `kvproto` package, we can't modify that package.

Reply:
ok